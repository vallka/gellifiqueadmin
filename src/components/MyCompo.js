import React from 'react';
import { TouchableOpacity, TouchableHighlight, Text, StyleSheet, View, Image } from 'react-native';

export class MyButton extends React.Component {
    render() {
      return (
          <TouchableHighlight style={styles.button} onPress={this.props.onPress} onLongPress ={this.props.onLongPress}>
            <Text style={styles.buttonText}>{this.props.text.toUpperCase()}</Text>
          </TouchableHighlight>
      );
    }
}
export class MyBackground extends React.Component {
  render() {
    return (
      <View
        style={{
          position: 'absolute',
          top: 0,
          left: 0,
          width: '100%',
          height: '100%',
        }}
      >
        <Image
          style={{
            flex: 1,
            resizeMode: 'cover',
            width: null,
            height: null,
            }}
          source={require ("../assets/images/bg.png")}
          />
      </View>
    );
  }
}

const styles = StyleSheet.create({
    button: {
      backgroundColor: '#C108C7',
      marginLeft: 40,
      marginRight: 40,
      marginTop: 20,
      marginBottom: 20,
      padding: 10,
      borderRadius: 10,
      alignItems: 'center',
  
    },
    buttonText: {
      color: '#fff',
      fontSize: 24,
    },
})  
