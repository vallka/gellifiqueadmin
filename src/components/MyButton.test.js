import React from 'react';
import {shallow} from 'enzyme';
import {MyButton} from './MyCompo';


describe('MyButton', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<MyButton text="test label"/>)
            expect(component).toMatchSnapshot()
        });
    });
});
