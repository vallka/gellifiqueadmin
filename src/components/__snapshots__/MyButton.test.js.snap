// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton Rendering should match to snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C108C7",
      "borderRadius": 10,
      "marginBottom": 20,
      "marginLeft": 40,
      "marginRight": 40,
      "marginTop": 20,
      "padding": 10,
    }
  }
  underlayColor="black"
>
  <Component
    style={
      Object {
        "color": "#fff",
        "fontSize": 24,
      }
    }
  >
    TEST LABEL
  </Component>
</TouchableHighlight>
`;
